invenio_accounts/views/rest.py

```diff
@@ -0,0 +1,533 @@
+
+    :param role: a new role to serialize into dict.
+    :return: dict from role.
+    :rtype: dict.
+    """
+    return dict(
+        id=role.id,
+        name=role.name,
+        description=role.description,
+    )
+
+
+def create_blueprint(app):
+    """Conditionally creates the blueprint."""
+    blueprint = Blueprint('invenio_accounts_rest_auth', __name__)
+
+    security_state = app.extensions['security']
+
+    if app.config['ACCOUNTS_REST_AUTH_VIEWS']:
+        # Resolve the view classes
+        authentication_views = {
+            k: obj_or_import_string(v)
+            for k, v in app.config.get('ACCOUNTS_REST_AUTH_VIEWS', {}).items()
+        }
+
+        blueprint.add_url_rule(
+            '/login',
+            view_func=authentication_views['login'].as_view('login'))
+        blueprint.add_url_rule(
+            '/logout',
+            view_func=authentication_views['logout'].as_view('logout'))
+        blueprint.add_url_rule(
+            '/me',
+            view_func=authentication_views['user_info'].as_view('user_info'))
+
+        if security_state.registerable:
+            blueprint.add_url_rule(
+                '/register',
+                view_func=authentication_views['register'].as_view('register'))
+
+        if security_state.changeable:
+            blueprint.add_url_rule(
+                '/change-password',
+                view_func=authentication_views['change_password'].as_view(
+                    'change_password'))
+
+        if security_state.recoverable:
+            blueprint.add_url_rule(
+                '/forgot-password',
+                view_func=authentication_views['forgot_password'].as_view(
+                    'forgot_password'))
+            blueprint.add_url_rule(
+                '/reset-password',
+                view_func=authentication_views['reset_password'].as_view(
+                    'reset_password'))
+
+        if security_state.confirmable:
+            blueprint.add_url_rule(
+                '/send-confirmation-email',
+                view_func=authentication_views['send_confirmation'].as_view(
+                    'send_confirmation'))
+
+            blueprint.add_url_rule(
+                '/confirm-email',
+                view_func=authentication_views['confirm_email'].as_view(
+                    'confirm_email'))
+
+        if app.config['ACCOUNTS_SESSION_ACTIVITY_ENABLED']:
+            blueprint.add_url_rule(
+                '/sessions',
+                view_func=authentication_views['sessions_list'].as_view(
+                    'sessions_list'))
+
+            blueprint.add_url_rule(
+                '/sessions/<sid_s>',
+                view_func=authentication_views['sessions_item'].as_view(
+                    'sessions_item'))
+    return blueprint
+
+
+class FlaskParser(FlaskParserBase):
+    """Parser to add FieldError to validation errors."""
+
+    # TODO: Add error codes to all messages (e.g. 'user-already-exists')
```

Comment:
TODO

Reply:
I'm not sure about this. I see that `FieldError` has an optional value [`code`](https://github.com/inveniosoftware/invenio-rest/blob/master/invenio_rest/errors.py#L31) (HTTP code) but I don't know where to take it from.
OTOH, `RESTValidationError` already [sets code](https://github.com/inveniosoftware/invenio-rest/blob/master/invenio_rest/errors.py#L107) as `400`.

Reply:
Issue created:
 - https://github.com/inveniosoftware/invenio-accounts/issues/305